quickwit/quickwit-search/src/find_trace_ids_collector.rs

```diff
@@ -0,0 +1,508 @@
+    ) -> tantivy::Result<Self::Child> {
+        let trace_id_ff_reader = segment_reader
+            .fast_fields()
+            .u64s(&self.trace_id_field_name)?;
+        let span_timestamp_column = segment_reader
+            .fast_fields()
+            .date(&self.span_timestamp_field_name)?;
+        let trace_id_field = segment_reader
+            .schema()
+            .get_field(&self.trace_id_field_name)?;
+        let inverted_index_reader = segment_reader.inverted_index(trace_id_field)?;
+        Ok(Self::Child {
+            trace_id_ff_reader,
+            span_timestamp_column,
+            inverted_index_reader,
+            select_trace_ids: SelectTraceIds::new(self.num_traces),
+        })
+    }
+
+    fn merge_fruits(
+        &self,
+        segment_fruits: Vec<<Self::Child as SegmentCollector>::Fruit>,
+    ) -> tantivy::Result<Self::Fruit> {
+        let mut trace_ids = Vec::with_capacity(self.num_traces);
+        let mut seen_trace_ids = FnvHashSet::default();
+
+        for trace_id in segment_fruits
+            .into_iter()
+            .map(BinaryHeap::from)
+            .map(binary_heap::IntoIterSorted::new)
+            .kmerge()
+        {
+            if !seen_trace_ids.contains(&trace_id.trace_id) {
+                seen_trace_ids.insert(trace_id.trace_id.clone());
+                trace_ids.push(trace_id);
+
+                if trace_ids.len() == self.num_traces {
+                    break;
+                }
+            }
+        }
+        Ok(trace_ids)
+    }
+
+    fn requires_scoring(&self) -> bool {
+        false
+    }
+}
+
+pub struct FindTraceIdsSegmentCollector {
+    trace_id_ff_reader: MultiValuedFastFieldReader<u64>,
+    span_timestamp_column: Arc<dyn Column<DateTime>>,
+    inverted_index_reader: Arc<InvertedIndexReader>,
+    select_trace_ids: SelectTraceIds,
+}
+
+impl FindTraceIdsSegmentCollector {
+    fn trace_id_term_ord(&self, doc: DocId) -> TermOrd {
+        self.trace_id_ff_reader
+            .get_first_val(doc)
+            .expect("There should be exactly one trace ID per span.")
+    }
+
+    fn span_timestamp(&self, doc: DocId) -> DateTime {
+        self.span_timestamp_column.get_val(doc)
+    }
+}
+
+impl SegmentCollector for FindTraceIdsSegmentCollector {
+    type Fruit = Vec<TraceIdFruit>;
+
+    fn collect(&mut self, doc: DocId, _score: Score) {
+        let term_ord = self.trace_id_term_ord(doc);
+        let span_timestamp = self.span_timestamp(doc);
+        self.select_trace_ids.collect(term_ord, span_timestamp);
+    }
+
+    fn harvest(self) -> Self::Fruit {
+        let mut trace_id_bytes = Vec::with_capacity(24);
+
+        self.select_trace_ids
+            .harvest()
+            .into_iter()
+            .map(|trace_id| {
+                let found = self.inverted_index_reader
+                    .terms()
+                    .ord_to_term(trace_id.term_ord, &mut trace_id_bytes)
+                    .expect("The term dict should be available.");
```

Comment:
The expect message is wrong...

ord_to_term returns an error if the term ordinal does not exist.

Reply:
What's the difference between the function returning an error and `Ok(false)`? How do we know the error is an actual io error vs. the term ord does not exist?